### DISCUSSION TITLE
Ender 6 and dgus-display connection

### DISCUSSION THREAD

**Discussion by FlatEarthDave** [Desuuuu/klipper#78] [Q&A]
I get this error, on my Ender 6, which is Creality board 2.3.1, when selecting  USART1, I believe it's PA9/PA10:

`Interface usart1 reserved for serial - can't reserve for uart_bus[usart1]`

I'm sure it's the same as I chose for the firmware before upgrading it. I tried every other connection, but they all just give me the normal trying to connect error.

**Comment by Desuuuu** [comment]
You're trying to use USART1 for both the host-communication and your display. Configure your display to use the correct usart interface.

**Comment by FlatEarthDave** [comment]
How do I do that exactly? I tried this:

`[dgus_display]
type: t5uid1
uart_bus: usart3
implementation: dgus_printer_menu
volume: 50
`

I get this:

`Internal error during ready callback: Timeout waiting for write acknowledgement
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Printer is shutdown`

I change to usar2, but I get this error:

`pin PA2 is reserved for usart2`

Sorry if it seems stupid, but I don't get it.

**Reply by FlatEarthDave** [reply]
Got it working. usart3 works, but I needed to update the core files, although I never got that "core update required" screen that I see in the main firmware. Thanks a bunch for all your hard work!

**Reply by Desuuuu** [reply]
Glad you got it figured out.

You should have been redirected to that screen indeed. I'll see if I can reproduce this issue, thanks.
